tests/test_openclhpp.cpp

```diff
@@ -3626,7 +3630,86 @@ void testDevice_GetInfo_CLDeviceName()
     clGetDeviceInfo_StubWithCallback(clGetInfo_testDeviceGetInfoCLDeviceName);
     cl::string deviceName = devicePool[0].getInfo<CL_DEVICE_NAME>();
     TEST_ASSERT_EQUAL_STRING(expected.c_str(), deviceName.c_str());
-} 
+}
+
+#if defined(cl_ext_device_fission)
+static cl_int clGetPlatformInfo_testDevice_createSubDevices(
+    cl_platform_id id, cl_platform_info param_name, size_t param_value_size,
+    void *param_value, size_t *param_value_size_ret, int num_calls) {
+    (void)num_calls;
+    const char *version = {"OpenCL 1.1 Mock"};
+    size_t bytes = strlen(version) + 1;
+    TEST_ASSERT_NOT_NULL(id);
+    TEST_ASSERT_EQUAL_PTR(make_platform_id(0), id);
+    TEST_ASSERT_EQUAL_HEX(CL_PLATFORM_VERSION, param_name);
+    TEST_ASSERT(param_value == nullptr || param_value_size >= bytes);
+    if (param_value_size_ret != nullptr)
+        *param_value_size_ret = bytes;
+    if (param_value != nullptr)
+        strcpy((char *) param_value, version);
+    
+    return CL_SUCCESS;
+}
+
+static cl_int clGetDeviceInfo_testDevice_createSubDevices(
+    cl_device_id id, cl_device_info param_name, size_t param_value_size,
+    void *param_value, size_t *param_value_size_ret, int num_calls) {
+    (void)id;
+    (void)num_calls;
+
+    TEST_ASSERT_EQUAL_HEX(CL_DEVICE_PLATFORM, param_name);
+    TEST_ASSERT(param_value == nullptr || param_value_size >= sizeof(cl_device_id));
+    if (param_value_size_ret != nullptr)
+        *param_value_size_ret = sizeof(cl_device_id);
+    if (param_value != nullptr)
+        *(cl_platform_id *)param_value = make_platform_id(0);
+    return CL_SUCCESS;
+    
+}
+
+static cl_int clCreateSubDevices_testDevice_createSubDevices(
+    cl_device_id in_device, const cl_device_partition_property *properties,
+    cl_uint num_devices, cl_device_id *out_devices, cl_uint *num_devices_ret,
+    int num) {
+    return 0;
+}
+
+static cl_int clCreateSubDevicesEXT_testDevice_createSubDevices(
+    cl_device_id device_in, const cl_device_partition_property_ext *properties,
+    cl_uint n, cl_device_id *out_devices, cl_uint *num, int num_calls) {
+  cl_int ret = CL_SUCCESS;
+  if (device_in == make_device_id(0)) {
+    return CL_SUCCESS;
+  } else if (device_in == make_device_id(1) && nullptr == num) {
+    return CL_DEVICE_NOT_FOUND;
+  } else {
+    return CL_DEVICE_NOT_FOUND;
+  }
+}
+
+void testDevice_createSubDevices() {
+  const cl_device_partition_property_ext properties =
+      CL_DEVICE_PARTITION_TYPES_EXT;
+  std::vector<cl::Device> devices;
+  cl_platform_id platform = make_platform_id(0);
+
+  clGetPlatformInfo_StubWithCallback(
+      clGetPlatformInfo_testDevice_createSubDevices);
+  clGetDeviceInfo_StubWithCallback(clGetDeviceInfo_testDevice_createSubDevices);
+  clCreateSubDevices_StubWithCallback(
+      clCreateSubDevices_testDevice_createSubDevices);
```

Comment:
I think this all can be replaced by:

```suggestion
    clGetDeviceInfo_StubWithCallback(clGetDeviceInfo_platform);
    clGetPlatformInfo_StubWithCallback(clGetPlatformInfo_version_1_1);
```

Then, many of the stub functions defined above are no longer needed (all but `clCreateSubDevicesEXT_testDevice_createSubDevices`).

Reply:
Changed